### ISSUE TITLE
Validation for Country / States field is run on initial page load

### ISSUE THREAD

**Issue by mikkamp** [woocommerce/woocommerce#20640]
**Is your feature request related to a problem? Please describe.**
When loading the Checkout page, either the Country or State field will be validated before the user interacted with the fields. The State field is validated if the country is preselected when the checkout is loaded. We can prevent this by setting the default location to "No location" (and making sure we sell to multiple countries). But if that is done then it's the Country field which runs the validation when loading the page.

When the validation is run it adds the classes `woocommerce-invalid` and `woocommerce-invalid-required-field` to the container element. On some themes this causes the label to turn red.

**Describe the solution you'd like**
It would be ideal if both the Country and the State field were only validated once the user interacted with the fields, and not on initial page load. Which will prevent the invalid classes from being added.

**Additional context**
The validation for these fields is hooked into the change event, see: https://github.com/woocommerce/woocommerce/blob/3.4.3/assets/js/frontend/checkout.js#L36

The following two change events are triggered for country fields on initial page load:
- https://github.com/woocommerce/woocommerce/blob/3.4.3/assets/js/frontend/checkout.js#L134
- https://github.com/woocommerce/woocommerce/blob/3.4.3/assets/js/frontend/country-select.js#L154

When the country is preselected on the checkout page (by either geo-location or limiting sales to one country) the following change event is what causes the state field to be validated on initial page load:
- https://github.com/woocommerce/woocommerce/blob/3.4.3/assets/js/frontend/country-select.js#L126

Is there maybe some way to determine if it's the init from the initial page load that is triggering these changes so it doesn't do the validation yet?

**Comment by lutrov** [comment]
If under general settings in admin we set the `Default customer location` to "Shop base address" things work as expected because the state and country fields get set to the store's country & state. So that's one way of overcoming the problem. But if we set the `Default customer location` to "No location by default" or "Geolocate", things don't work properly.

**Comment by peterfabian** [comment]
I would be inclined to simply remove https://github.com/woocommerce/woocommerce/blob/3.4.3/assets/js/frontend/country-select.js#L154 and https://github.com/woocommerce/woocommerce/blob/3.4.3/assets/js/frontend/checkout.js#L134 as I haven't found any side effects to simply not triggering `change()` there, but maybe I'm missing something? 
@mikejolley what do you think?

I checked the functions linked to `changed` event on billing country and see these being triggered:
- country to state code: but as far as I can see, the country/county/state inputs are loaded correctly even without this function running: https://github.com/woocommerce/woocommerce/blob/3.4.3/assets/js/frontend/country-select.js#L82
- password strength meter -- I assume this does not need to run on load
- ship_to_different_address -- as far as I can see, this is something user needs to check a checkbox first to activate 
- maybe_input_changed -- which triggers `input_changed` if `wc_checkout_form.dirtyInput` is truthy, which (I assume) should not be the case right after the on load, but even if it is called from here, it's the same as next item
- input_changed -- this might actually call `trigger_update_checkout` if any of the address fields are pre-filled, but that function just resets update timer, clears `dirtyInput` flag and then triggers `update_checkout`, which would [be called anyway from `init_checkout`](https://github.com/woocommerce/woocommerce/blob/3.4.3/assets/js/frontend/checkout.js#L134).
- validation mentioned by @mikkamp

**Comment by mikejolley** [comment]
https://github.com/woocommerce/woocommerce/pull/20655 does indeed seem to work. Let's get that finished and ready for review and go with that 👍 cc @claudiulodro 

See https://github.com/woocommerce/woocommerce/pull/20655

**Comment by gingerlime** [comment]
> I haven't found any side effects to simply not triggering `change()` 

I spotted a problem on #21835 related to e98694b8406834ba01bbdb16622e485c81f50115 and 11da1898b83832057b01eec921c5e881c7cdc509. The first thing on #21835 is that the states dropdown isn't shown on first page load. But also I'm having a different problem. I'm still not entirely sure what's going on, but I'm using a plugin that adds a couple of address fields. It worked before this change, but after it, on the first visit to the checkout page, the whole Country sub-section appears in the wrong place (on top, before the name fields). After reloading the page, things somehow drop into place and the country moves down... I can share a short video or some screenshots if it helps, but since the site is in Japanese, it's a bit hard to follow which fields move where. 

I can load the states manually using javascript and running `jQuery( ':input.country_to_state' ).change();` and the states would be loaded, but this is independent of the order of fields, which only resolves itself on reload.
